Type error with Typescript 4.8

The following template parameter is throwing an error with TypeScript 4.8 because the template of `Conditional` type is restricted to a subset `Conditional<CD extends FieldDef<any> | DatumDef | ValueDef<any> | ExprRef | SignalRef>`

https://github.com/vega/vega-lite/blob/c97e30a8da569aabab7f348673899f43aa3b8825/src/channeldef.ts#L158

The error raised is 

```
  ../node_modules/vega-lite/build/src/channeldef.d.ts(63,66): error TS2344: Type 'T' does not satisfy the constraint 'SignalRef | ExprRef | FieldDef<any, any> | DatumDef<string, SignalRef | DateTime | ExprRef | PrimitiveValue> | ValueDef<...>'.
    Type 'T' is not assignable to type 'DatumDef<string, SignalRef | DateTime | ExprRef | PrimitiveValue>'.
```

Seen in https://github.com/jupyterlab/jupyterlab/actions/runs/3217544851/jobs/5260639447

--- a/src/channeldef.ts
+++ b/src/channeldef.ts
@@ -143,9 +143,7 @@ export type TypeForShape = 'nominal' | 'ordinal' | 'geojson';
 
 export type ConditionalTemplate = FieldDef<any> | DatumDef | ValueDef<any> | ExprRef | SignalRef;
 
-export type Conditional<CD extends ConditionalTemplate> =
-  | ConditionalPredicate<CD>
-  | ConditionalParameter<CD>;
+export type Conditional<CD extends ConditionalTemplate> = ConditionalPredicate<CD> | ConditionalParameter<CD>;
 
 export type ConditionalPredicate<CD extends ConditionalTemplate> = {
   /**
@@ -154,8 +152,7 @@ export type ConditionalPredicate<CD extends ConditionalTemplate> = {
   test: LogicalComposition<Predicate>;
 } & CD;
 
-export type ConditionalParameter<CD extends ConditionalTemplate> =
-  ParameterPredicate & CD;
+export type ConditionalParameter<CD extends ConditionalTemplate> = ParameterPredicate & CD;
 
 export function isConditionalParameter<T extends ConditionalTemplate>(c: Conditional<T>): c is ConditionalParameter<T> {
   return c['param'];

--- a/src/channeldef.ts
+++ b/src/channeldef.ts
@@ -141,21 +141,23 @@ export type StringValueDefWithCondition<F extends Field, T extends Type = Standa
 >;
 export type TypeForShape = 'nominal' | 'ordinal' | 'geojson';
 
-export type Conditional<CD extends FieldDef<any> | DatumDef | ValueDef<any> | ExprRef | SignalRef> =
+export type ConditionalTemplate = FieldDef<any> | DatumDef | ValueDef<any> | ExprRef | SignalRef;
+
+export type Conditional<CD extends ConditionalTemplate> =
   | ConditionalPredicate<CD>
   | ConditionalParameter<CD>;
 
-export type ConditionalPredicate<CD extends FieldDef<any> | DatumDef | ValueDef<any> | ExprRef | SignalRef> = {
+export type ConditionalPredicate<CD extends ConditionalTemplate> = {
   /**
    * Predicate for triggering the condition
    */
   test: LogicalComposition<Predicate>;
 } & CD;
 
-export type ConditionalParameter<CD extends FieldDef<any> | DatumDef | ValueDef<any> | ExprRef | SignalRef> =
+export type ConditionalParameter<CD extends ConditionalTemplate> =
   ParameterPredicate & CD;
 
-export function isConditionalParameter<T>(c: Conditional<T>): c is ConditionalParameter<T> {
+export function isConditionalParameter<T extends ConditionalTemplate>(c: Conditional<T>): c is ConditionalParameter<T> {
   return c['param'];
 }
 

